docs/development_guide.md

```diff
@@ -0,0 +1,121 @@
+Development Guide
+-----------------
+
+This guide is for those that want to help hack on the Slack bridge, and need some instruction on how
+to set it up. This is **not** a guide for setting up an instance for production usage.
+
+For information on how to contribute issues and or pull requests, please read [CONTRIBUTING](../CONTRIBUTING.md).
+
+## Setting up your environment
+
+This section explains how to setup Synapse, Riot and the Slack bridge for local development.
+
+### Prereqs 
+
+Ensure at the minimum you have installed:
+
+- NodeJS ( [nvm](https://github.com/nvm-sh/nvm) is a good tool for this)
+- Docker (optional, but it will make your life easier)
+
+For the sake of making this easier to follow, we will create a directory in the home directory
+called `slack-bridge-env`:
+
+```bash
+mkdir ~/slack-bridge-env
+cd ~/slack-bridge-env
+```
+
+### Setting up Synapse
+
+Largely to setup Synapse, you can follow https://hub.docker.com/r/matrixdotorg/synapse and just ensure your data directory points to `~/slack-bridge-env/synapse`.  
```

Comment:
There's a change of voice/tone here which is a little jarring: the previous section uses a voice where the guide is the user ("we", "our", etc) but here it changes to be instructional ("do this", "you", etc). Personally I prefer the more fuzzy feeling of the guide saying "we", "us", "our", and such.

Reply:
The docker commands in the hub.docker.com docs need to be changed. I recommend including them explicitly here with the needed changes, for example:

```
docker run -it --rm \
    -v ~/slack-bridge-env/synapse:/data \
    -e SYNAPSE_SERVER_NAME=localhost \
    -e SYNAPSE_REPORT_STATS=no \
    matrixdotorg/synapse:latest generate

docker run -d --name synapse \
    -v ~/slack-bridge-env/synapse:/data \
    -p 8008:8008 \
    matrixdotorg/synapse:latest
```